### ISSUE TITLE
Reduce the number of special variables bound in quilc::process-options

### ISSUE THREAD

**Issue by appleby** [rigetti/quilc#226]
Per review comments in #212, it would be nice to reduce the number of special variables in `app/src/entry-point.lisp` wherever that is possible and makes sense.

Most of the bindings occur in `quilc::process-options`.

**Comment by notmgsk** [comment]
Can we close this now?

**Comment by appleby** [comment]
No, this is a (perhaps poorly worded) separate issue to the one addressed in #212. This is about getting rid of the special vars altogether wherever possible, whereas #212 was about moving away from global mutation via `setf` and friends to dynamic let bindings.
